### ISSUE TITLE
Detects winter (heating) and summer (cooling) modes?

### ISSUE THREAD

**Issue by marcospg75** [asev/homeassistant-uponor#16]
I'm waiting for change my uponor system for a new one that support mobile App.
I wonder if this custom_integration supports both modes (cooling and heating)

Best Regards

**Comment by thecem** [comment]
I don’t now how this is implemented in  the api of r-208.
May you have to try.

**Comment by marcospg75** [comment]
I was seing your code... and I guess is another HVAC_MODE.

    @property
    def hvac_modes(self):
        return [HVAC_MODE_HEAT, HVAC_MODE_OFF]

perhaps HVAC_MODE_COOL
That changes the behavior on the climate doing to turn on when room's temperature is greater than the (instead of lower than) the set one.

**Comment by marcospg75** [comment]
Hello,
I have now installed the R208 with both, heating and cooling modes with the App.
How can I help you (or you can help me) to add this functionality to the custom component?
Best Regards

**Comment by asev** [comment]
Hello.
The only data that this integration sends to Uponor controller is temperature setpoint.

HVAC mode that you see has very basic logic. `HVAC_MODE_OFF` when temperature set to minimum (by default 5℃) otherwise it's `HVAC_MODE_HEAT`.
When you turn off climate entity it just sets temperature to the minimum. When you turn on - it sets temperate to the last known (#7).

I see no reason why it would not work for cooling. The only problem that in Home Assistant it will say "Heating" :)
Would you be able to test cooling at this time of a year if I prepare some adjustments?

**Comment by marcospg75** [comment]
Sure! I can test cooling mode!! just ask what you need (and perhaps some detailed instructions to get it).
Keep in mind that MODE_COOL changes the behaviour of all the thermostats (not only saying "Heating") because the will activate when temperature of the room is higher than the thermostat temperature (instead of lower for activating heating mode).
IMHO it would be better to match themostats with Dual Mode Generic Thermostat custom-component (https://github.com/zacs/ha-dualmodegeneric) than the generic_thermostat platform from Home Assistant.
Please don't hesitate in asking for all the test you need :)

**Comment by marcospg75** [comment]
I did some initial tasks :)
When cooling mode is activated this variable:
payload.output.vars[31].waspVarName sys_heat_cool_mode is set to 1 (0 in heating mode)

**Comment by thecem** [comment]
I think with `payload.output.vars[31].waspVarName sys_heat_cool_mode is set to 1 ` the logic of thermostat will be changed in the x-2(4|6)5 to cooling and switch on the configured and connected to the control modul heat cool system only if the temp is higher.

**Comment by asev** [comment]
I would need to figure out few things:
1. How to identify that system supports cooling? My guess `sys_cooling_available`. Is it set to 1 in your system?
2. How to switch to cooling mode?  Already answered `sys_heat_cool_mode` ✅
3. Is there a way to say that specific thermostat is in cooling mode? My guess `C1_T1_heat_cool_mode` or `C1_T1_stat_cb_heat_cool_mode`. Are those set to 1 when you activate cooling mode?
4. How to tell that actual cooling is happening - cooling actuators are open? For heating it's `C1_T1_stat_cb_actuator`. And my guess that `C1_T1_stat_cb_sub_actuator` represents cooling.

**Comment by marcospg75** [comment]
Answers inline:

1. How to identify that system supports cooling? My guess `sys_cooling_available`. Is it set to 1 in your system? Right ✅
2. How to switch to cooling mode?  Already answered `sys_heat_cool_mode` ✅
3. Is there a way to say that specific thermostat is in cooling mode? My guess `C1_T1_heat_cool_mode` or `C1_T1_stat_cb_heat_cool_mode`. Are those set to 1 when you activate cooling mode?
C1_stat_heat_cool_mode =1
C1_T1_stat_cb_heat_cool_mode = 1 (in general CxTz_stat_cb_heat_cool_mode = 1) ✅
4. How to tell that actual cooling is happening - cooling actuators are open? For heating it's `C1_T1_stat_cb_actuator`. And my guess that `C1_T1_stat_cb_sub_actuator` represents cooling.
C1_T1_stat_cb_actuator =1 (in general CxTz_stat_cb_actuator =1) ✅


Atached you can find 2 files (cooling.txt and heating.txt) so you can find all the difference between both modes.
[cooling.txt](https://github.com/asev/homeassistant-uponor/files/5893059/cooling.txt)
[heating.txt](https://github.com/asev/homeassistant-uponor/files/5893060/heating.txt)

If you need something more, here I am to help.

**Comment by asev** [comment]
@marcospg75 I've released v0.7.0. Could you try to update and test it?

**Comment by marcospg75** [comment]
I will test later this week!
Thank you very much for the quick version supporting cooling :)

**Comment by marcospg75** [comment]
Finally, I couldn't wait :)
I'm very happy with the 2 new switches!! 🥇 
Both are running well, and the official app changes with a 5sec delay (which is good enougth for this purpose).
But... I notice that the HA thermostat entities don't update their set temperture.
With the app, you can manage 2 offsets of temperature:
* One for ECO (2ºC in my case) --> minus when ECO-Heating, plus when ECO-Cooling
* Another one for Cooling (5ºC in my case) --> plus when moving to cooling, minus when mooving to heating

I can see this behaviour in the Uponor App, but seems like HA thermostates don't refresh this information.
In fact, i tested to get all information of the thermostat in Node-Red and it gives me the "non-updated" temperature.

**Comment by marcospg75** [comment]
More information about this:
I think is not a sync or refresh issue. It's just the way they apply the offsets.
Example:
If you set the temperature to 20ºC and check AWAY Mode (2ºC in my case), in the App you can see 18ºC set but in HA you see 20ºC
Now, if you edit the temperature on HA thermostat and set to 21ºC, in de App you will see 19ºC!!!
and viceversa... if you set in the App manually to 15ºC, in HA the thermostat shows 17ºC

So, my opinion is that the API keep sending the original set temperature but also with the right offset to show the right final temperature in the App... and, this should be done in the integration for both offsets (cooling and ECO).

I hope I was clear in my explanation :)

**Comment by asev** [comment]
Thanks for the feedback. I've noticed that as well. I've released v0.7.1. This issue should be fixed there.

**Comment by marcospg75** [comment]
Hello,
Some questions and some issues...
Question: Why are defined 3 presets? In the App there is just AWAY/NOT AWAY... I figure out that this is equivalent to AWAY/CONFORT... but... there is a mode ECO... how it works?

Issue 1: The ECO offset is applied when activate Uponor Away Switch... but, always decrementing the temperture. Keep in mind that in cooling mode, the offset should be added

Issue 2: The Cooling Offset (5ºC in my case) is not applied when changing from heating to cooling and viceverse

Issue 3: When a thermostat is OFF, temperature is set to 5C (due to API limitations). THis is true in heating mode. In Cooling mode, OFF thermostats should be set to 35ºC

Issue 4: I notice that you apply ECO offset also to OFF thermostat... so, if Uponor Away is ON, and a thermostat is OFF... when you switch off Uponor Away, the temperature of this thermostat is set to 7ºC.... and should be kept in 5C because is OFF (the Uponor Offical App is doing this in the right way).

Thank you very much for your support... and please ask me for any test or log you need.

**Comment by thecem** [comment]
ECO: This one will be shown in HA if you sign a room to one of the preset ECO-Profile, but not if you use a custom defined ECO-Profile. So I think the custom ECO-Profile is actual not in the integration.
At least for heating the ECO (without custom Profiles) /COMFORT/AWAY setting is working.
COMFORT: Is there a reason why comfort and not "home"? Comfort funktion in Uponor is to have the floor with a static minimal temp setted in % to have water flow in the floor, to prevent cold floors if you heat with a additional heater or a stove. Not that this could be confound. I would vote for home since this is the standard of the climate entity.

Thanks a lot!

**Comment by marcospg75** [comment]
So, if I understood... you always should be in one of this 3 profiles: ECO, AWAY or CONFORT... right?
From my side, as I use HA and NodeRed to automate and define schedules, I won't use at all the ECO-Profile from the App.

But, It would be great if you can add the 2 offsets in the integration (ECO and COOLING Offsets). Whith these features, I should remove the Uponor App forever :)

Best Regards

**Comment by thecem** [comment]
Beside that, a question to the app. Do you have also a problem with the resolution of the latest App. Since the last changed it is really hard to get the menu (upper left) pushed, I use a 12 max and it is up to not possible to get this Menu opened. I wrote Uponor but no one answered up to now.

**Comment by marcospg75** [comment]
> Beside that, a question to the app. Do you have also a problem with the resolution of the latest App. Since the last changed it is really hard to get the menu (upper left) pushed, I use a 12 max and it is up to not possible to get this Menu opened. I wrote Uponor but no one answered up to now.

I use a Samsung Galaxy S10, so it's different App. I have no issues with that

**Comment by asev** [comment]
@thecem is correct about ECO preset. It's activated when scheduled ECO profile is on OR Temporary ECO mode activated on the mobile app.

>  I would vote for home since this is the standard of the climate entity.

I've removed this preset. Now it has only Eco and Away.

@marcospg75 I've fixed those issues in v0.7.2. Check if it works with actual cooling. :)

> But, It would be great if you can add the 2 offsets in the integration (ECO and COOLING Offsets). Whith these features, I could remove the Uponor App forever :)

I plan to add a service to the integration. So you would be able to send any variable to API.
